### ISSUE TITLE
Environmental variables must never be used in production

### ISSUE THREAD

**Issue by narvikd** [narvikd/victor_robles_bot#2]
Right now Vault's model is implemented in a very crude way:

The keys are being rotated on a weekly basis, but we're still accessing the secrets via an environmental variable that can be easily dumped.

The secrets and Vault itself must be stored on another server, the access to it should not occur automatically since we just need the secret key to log in.

Right now only Vault is on another server, but the secrets are passed to the program and they're stored as an env.

One may think that an attacker will have a very hard time dumping memory from a shell. But this is not the case, without any external programs you could easily do:
```
#!/bin/bash
cat /proc/$1/maps | grep "rw-p" | awk '{print $1}' | ( IFS="-"
while read a b; do
    dd if=/proc/$1/mem bs=$( getconf PAGESIZE ) iflag=skip_bytes,count_bytes \
       skip=$(( 0x$a )) count=$(( 0x$b - 0x$a )) of="$1_mem_$a.bin"
done )
```
And it's dumped.

I'm still investigation the best approach to fix this.

**Comment by narvikd** [comment]
After some deliberation, I concluded that I really didn't like the idea I was proposing: Passing secrets via yet another program, and by such adding another attack surface into the mix.

So instead of reinventing the wheel, I looked up how we have been doing this for decades:
**stdin**.

With environ a file exists for each running process. Any local code execution exploit, even without privileges, can exploit it, it is very simple to do, it's just a file.

If I pipe things into the stdin, an outside process can only intercept the messages by attaching to it. It is more complex, besides, you can configure Linux to prevent an unprivileged user/process from calling the system debugger:
Even if it's another process started by the same user that is executing this program.

It is not perfect, but in almost all cases it's much much better than doing things like:
* Calling env
* cat mypass.txt | myprogram
* echo mypass | myprogram

Note: I know echo is just builtin on the shell and it's then written to stdin, but if it's invoked incorrectly it will result in a big security risk.

**TLDR:** stdin is not a perfect solution but it has a much smaller attack surface area, and it makes it so much better than passing env params or adding complexity. We could implement something to treat the key passed by stdin but I don't think we need it.
